File: python/nt_solutions/ml_4_sgd/exo0.py
q=20  # Number of paths computed in parallel.
niter = 1000
x = np.zeros( (q,niter) )
x[:,0] = np.random.rand(q,1).flatten()-.5 # Initial conditions.
for i in np.arange(1,niter):
    u = np.random.rand(q,1)>.5
    xx = MakeCol(x[:,i-1])
    tau = 1.0/(10+i)
    x1 = xx - tau * ( u*df(xx,1) + (1-u)*df(xx,2) )
    x[:,i] = x1.flatten()
# display
plt.clf
plt.plot(x.transpose(), 'r')
plt.axis('tight')

File: r/nt_solutions/ml_4_sgd/exo0.R
q = 20  # Number of paths computed in parallel.
niter = 1000

x = matrix(0, q, niter)
x[,0] = randn(q,1) - .5 # Initial conditions.

for (i in 2:niter)
{
    u = randn(q,1) > 0.5
    xx = x[, i - 1]
    tau = 1.0 / (10 + i)
    x1 = xx - tau * (u * df(xx,1) + (1-u) * df(xx,2))
    x[,i] = x1
}


matplot(t(x), col='red', type="l", xlab="", ylab="")
